apps/nx-workshop-e2e/tests/nx-workshop.spec.ts

```diff
@@ -87,6 +87,44 @@ describe('nx-react-workshop', () => {
         }
       );
 
+      /**
+       * TODO: figure out what best approach for getProjects(tree) is
+       *
+       * Right now, if you call the removeGenerator for a project and then
+       * immediately call getProjects(tree), getProjects will throw.
+       *
+       * This is due to getProjects fetching a list of project.json files
+       * using globWithWorkspaceContextSync, which globs against the real
+       * filesystem rather than the in-memory tree.
+       *
+       * Once the list of project.json flies is fetched, the project details
+       * are they attempted to be fetched from the in-memory tree, which
+       * subsequently fails because we just deleted it.
+       *
+       * See: {@link https://github.com/nrwl/nx/blob/master/packages/nx/src/generators/utils/project-configuration.ts#L251}
+       *
+       * Re-enabling this test will recreate the issue as long as the current
+       * workspace has at least one project in it (which the previous test will create)
+       */
```

Comment:
I've reached out to @FrozenPandaz about this and will address it in a future update once we figure out what to do with `getProjects(tree)`.

For now, we just need to run the "reset workspace" migration separately from all the rest.